### PR TITLE
Room settings-Members added/deleted in a room in offline mode are not grayed out/crossed

### PR DESCRIPTION
**Pull request by tienifr**
### Fixed Issues

$ https://github.com/Expensify/App/issues/31764
PROPOSAL: https://github.com/Expensify/App/issues/31764#issuecomment-1823770551

### Tests

1. Login with any account
2. Open any room (or create it) -> Members
3. Turn off the internet connection
4. Invite some new members
5. Delete some members
6. Verify that members added offline are grayed out
7. Verify that members removed offline are crossed out

- [x] Verify that no errors appear in the JS console

### Offline tests

Same as above

### QA Steps

1. Login with any account
2. Open any room (or create it) -> Members
3. Turn off the internet connection
4. Invite some new members
5. Delete some members
6. Verify that members added offline are grayed out
7. Verify that members removed offline are crossed out

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
        - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/113963320/7003c3fb-3b83-429c-a771-97f47cfa1be3

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/113963320/2bc6abd5-b35d-4a71-8d05-2ba369a7dc68

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/113963320/e2f82e15-b0a3-4a43-bedc-9be0a9852a9d

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/113963320/b45e2eea-2cb9-4672-bf47-589c29e4c139

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/113963320/10d38013-351d-4847-8e26-8df12df6d4b3

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/113963320/96e4b1d8-e86c-4411-bf6f-51e367beb453

</details>

### CODE REVIEW

**Comment by alitoshmatov** [comment]
Will take a look today

**Comment by alitoshmatov** [comment]
Sorry This one went under my radar, @tienifr I don't have access to expensifail account, is this required?

**Comment by alitoshmatov** [comment]
@Bump @tienifr

File: src/languages/es.ts
Comment (alitoshmatov) [inline src/languages/es.ts:1941]: Why is this empty?

File: src/languages/en.ts
Comment (alitoshmatov) [inline src/languages/en.ts:1915]: This sounds a little bit of, especially `that room member` part. Did someone confirm this texts?

**Comment by alitoshmatov** [comment]
@tienifr Yes, I mean we need to confirm text for both languages

File: src/languages/en.ts
Comment (deetergp) [reply src/languages/en.ts:2077]: I think what you have here is fine.

Comment (deetergp) [reply src/languages/en.ts:2077]: Actually, let's add the `Waiting for copy` label and get official @Expensify/marketing sign-off.

Comment (jamesdeanexpensify) [reply src/languages/en.ts:2077]: Ah, I don't think the "Waiting for copy" label works on pull requests! It looks like this is for a generic error message and we can't drill down into more specific reasoning?

**Comment by deetergp** [comment]
@tienifr Just a heads up that you've got merge conflicts. Not sure if it's better to resolve them now wait till after marketing reviews the error copy.

File: src/languages/en.ts
Comment (jamesdeanexpensify) [reply src/languages/en.ts:2077]: Also, where will these messages show up exactly and how will they look? 

Not knowing much else, I'd suggest slight tweaks:
- There was a problem adding this member to the room.
- There was a problem removing this member from the room.

**Comment by deetergp** [comment]
@tienifr @alitoshmatov Should we get an ES re-translation of the updated EN text? Also, there appear to be failing TypeScript checks.

File: src/languages/en.ts
Comment (jamesdeanexpensify) [reply src/languages/en.ts]: Getting a check [here](https://expensify.slack.com/archives/C21FRDWCV/p1708715288435469).

Comment (jamesdeanexpensify) [reply src/languages/en.ts]: This is already in the code it seems. I'm not an engineer, but an engineer confirmed it. @deetergp are you able to see the conversation [here](https://expensify.slack.com/archives/C21FRDWCV/p1708715288435469)? Thanks!

File: src/languages/en.ts
Comment (deetergp) [inline src/languages/en.ts:2086]: Delete this. Instead, use the existing messages at `workspace.people.error`

```suggestion
```

File: src/libs/actions/Report.ts
Comment (deetergp) [inline src/libs/actions/Report.ts:2452]:
```suggestion
            errors: ErrorUtils.getMicroSecondOnyxError('workspace.people.error.genericRemove'),
```

File: src/languages/es.ts
Comment (deetergp) [inline src/languages/es.ts:2114]:
```suggestion
```

File: src/libs/actions/Report.ts
Comment (deetergp) [inline src/libs/actions/Report.ts:2389]:
```suggestion
            errors: ErrorUtils.getMicroSecondOnyxError('workspace.people.error.genericAdd'),
```

**Review by deetergp — CHANGES_REQUESTED** [review]

**Comment by deetergp** [comment]
@tienifr I could swear I responded to you yesterday, but apparently never hit the big green `Comment` button after typing it up. Sorry! Anyway, we discussed it in Slack and decided using the existing _workspace_ verbiage would be fine.

**Comment by alitoshmatov** [comment]
Working on it

**Comment by alitoshmatov** [comment]
@tienifr I am a bit confused, when I am testing removed users are disappearing from the list not being crossed out

https://github.com/Expensify/App/assets/59907218/f6b52f63-e694-45ba-b7f5-060b9d6869e1

**Comment by alitoshmatov** [comment]
Is this intentional behavior change or is it really a bug? If latter is true we should hold and address this bug

**Comment by deetergp** [comment]
Confirmed that it works like that in `main` and definitely does not feel like intended behavior. It feels like something we ought to be able to fix here, doesn't it?

**Comment by deetergp** [comment]
That sounds reasonable to me. @alitoshmatov what do you think?

**Comment by alitoshmatov** [comment]
Okay let's go with this

**Comment by alitoshmatov** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/59907218/561e620f-9418-49f0-b335-e4ca122c2bc2




</details>

<details>
<summary>Android: mWeb Chrome</summary>



https://github.com/Expensify/App/assets/59907218/c2093be9-ca59-4e6d-ad5b-b9a85aa455b1




</details>

<details>
<summary>iOS: Native</summary>



https://github.com/Expensify/App/assets/59907218/99f40c11-02b8-4273-bdf3-4dc596a47231




</details>

<details>
<summary>iOS: mWeb Safari</summary>


https://github.com/Expensify/App/assets/59907218/87f779b0-b2a4-4bd6-875f-a0c96bdf11a9




</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/59907218/4b174d8f-590d-4e59-b570-68a16eb0800e






</details>

<details>
<summary>MacOS: Desktop</summary>



https://github.com/Expensify/App/assets/59907218/ed14cf8f-86a9-40a6-9829-dee6c287bc05


</details>

**Review by alitoshmatov — APPROVED** [review]
LGTM!

**Comment by deetergp** [comment]
@tienifr Looks like you've collected a few merge conflicts. Once you sort those out, we'll get a final review in and ship this puppy 🚀

**Comment by alitoshmatov** [comment]
@tienifr I agree we should still apply strikethrough on delete because it is part of the main issue

**Comment by deetergp** [comment]
It looks to me like strikethrough is already a part of https://github.com/Expensify/App/pull/35883. Is there anything more to do here?

**Comment by deetergp** [comment]
@tienifr Okay, if it's still broken on `main` even after https://github.com/Expensify/App/pull/35883 was merged, let's go ahed and address it here. 👍

**Comment by alitoshmatov** [comment]
@tienifr It is still not working on me. Can you recheck

https://github.com/Expensify/App/assets/59907218/b27d98d8-03f0-4152-9f87-cc2a28f277f4

**Review by alitoshmatov — APPROVED** [review]
Okay, code looks good to me, we can finally merge it. cc: @deetergp 

https://github.com/Expensify/App/assets/59907218/07fe7c40-2381-4bb7-bcff-7ba5dc1e80ed

**Review by deetergp — APPROVED** [review]
Code looks good and tested well for me on web. Thanks for seeing it through 👍

**Comment by kavimuru** [comment]
The bug #31764 is still reproduced in all the platforms. 
https://platform.applause.com/services/links/v1/external/81c6f4fe4b9036315cc36e64a106379d87357d039c35cbe9d9b91daf3932d50e
https://platform.applause.com/services/links/v1/external/eebecd0e7ff3809fd7a51fbc1ba361563c6a0de1fd0dee48d28f1703fc814d5c

**Pull request merged**
